filebeat/registrar/registrar.go

```diff
@@ -85,6 +87,12 @@ func (r *Registrar) Init() error {
 		return fmt.Errorf("Registry file path is not a regular file: %s", r.registryFile)
 	}
 
+	// Update registry file permissions
+	err = os.Chmod(r.registryFile, r.registryFilePermission)
```

Comment:
Not sure if we really should update the file permission here. I would rather suggest to change in on line 262 so we already write it with the correct permission.